### DISCUSSION TITLE
Cannot see generated SQL

### DISCUSSION THREAD

**Discussion by dgmoex** [linq2db/linq2db#3413] [Q&A] [answered]
Hello
I upgrade my app from net core 3.1 to 6 and linq2db from 2.9.8 to 3.6.0 
Some queries return antoher result and break my autotests, I want see SQL that was generated. 
I set TurnTraceSwitchOn and WriteTraceLine in database context, run **Debug** autotest. Query is simple and executed successful, but I not see generated sql on Output-Debug. 
I reset source code to net core 3.1 and linq2db 2.9.8 and this work fine and print SQL. What happen and how I can see generated SQL?

```
public class Database : DataConnection
{
        internal static readonly Logger Log = LogManager.GetLogger(typeof(Database).FullName);

        internal Database(IDataProvider provider, IDbConnection connection) : base(provider, connection)
        {
            Debug.WriteLine(" test print 1"); // always printed!
            
            LinqToDB.Data.DataConnection.TurnTraceSwitchOn();
            LinqToDB.Data.DataConnection.WriteTraceLine = (message, displayName, traceLevel) =>
            {
                Debug.WriteLine($"{displayName} {message}"); // never printed!
            };
        }
...
}
```
Visual Studio 2022 Professional

**Comment by MaceWindu** [comment]
Unsure why you don't have logs - this functionality used a lot by our tests without any issue. I personally use same approach for logging (but feed data to logger instead of `Debug` helper).

BTW, if you set global logger (`WriteTraceLine`) - you don't need to initialize it on each connection creation.

**Reply by dgmoex** [reply]
i moved this code to fabric, that produce instance of context, and it work. Mysteriously..
thanks for reply,  regards

**Reply by Ben290182** [reply, marked as answer]
If someone also comes to this question wondering why their trace is not working, for me it was I had to call TurnTraceSwitchOn and WriteTraceLine before I instantiated my DataConnection.
